src/CopyrighFixer/CopyrighFixer/signer.cpp

```diff
@@ -7,13 +7,55 @@
 
 #include "signer.h"
 #include "CopyrighFixer/ifilemanager.h"
+#include <QDir>
 
 namespace CopyrighFixer {
 Signer::Signer() {
 
 }
 
-bool Signer::checkSign(const Config &objConf) {
+bool Signer::checkSign(const Config &objConf) {    
+
+    Config currConfig = objConf;
+
+    QDir currentFolder(currConfig.getSrcDir());
+    currentFolder.setFilter(QDir::Dirs | QDir::Files);
+
+    QFileInfoList folderItems(currentFolder.entryInfoList());
+
+    for (auto i_file: folderItems) {
+
+        if (i_file.fileName() == "." || i_file.fileName() == "..") {
+            continue;
+        }
```

Comment:
please use the [NoDotAndDotDot](https://doc.qt.io/qt-5/qdir.html#Filter-enum) atribute instead custom validation.